pywr/domains/river.py

```diff
@@ -95,9 +107,138 @@ def __init__(self, model, *args, **kwargs):
         else:
             # reinstate the given cost parameter to pass to the parent constructors
             kwargs["cost"] = cost
+
+        bathymetry = kwargs.pop('bathymetry', None)
+        volume = kwargs.pop('volume', None)
+        level = kwargs.pop('level', None)
+        area = kwargs.pop('area', None)
+        self.weather_cost = kwargs.pop('weather_cost', -999)
+        self.evaporation_cost = kwargs.pop('evaporation_cost', -999)
+        self.rainfall_cost = kwargs.pop('rainfall_cost', -999)
+        const = kwargs.pop('const', 1e6 * 1e-3 * 1e-6)
+
+        # Pywr Storage does not expect a 'weather' kwargs, so move this to instance
+        self.weather = kwargs.pop("weather", None)
+
+
         super(Reservoir, self).__init__(model, *args, **kwargs)
 
 
+        self.const = ConstantParameter(model, const)
+
+        self.rainfall_node = None
+        self.rainfall_recorder = None
+        self.evaporation_node = None
+        self.evaporation_recorder = None
+
+    @classmethod
+    def pre_load(cls, model, data):
+
+        bathymetry = data.pop("bathymetry", None)
+        name = data.pop("name")
+        node = cls(name=name, model=model, **data)
+
+        if bathymetry is not None:
+            if isinstance(bathymetry, str):
+                bathymetry = load_parameter(model, bathymetry)
```

Comment:
This can use the Pywr storage node 'area' and 'level' parameters. Area and Level should be defined indenpendently rather than in the same table as often the area and level dataframes dont have the same number of rows.